### ISSUE TITLE
Finish the SpinOp implementation

### ISSUE THREAD

**Issue by manoelmarques** [Qiskit/qiskit-nature#22]
Migrated from Enterprise Github.
Creator: @mrossinek

- [ ] implement the `to_opflow()` methods for both, `primitives.SpinOperator` and `SpinSumOp`
- [ ] unittest all methods

**Comment by manoelmarques** [comment]
Creator: @mrossinek

Below is Simon's original code for the `to_qiskit()` mapping which needs to be rewritten to use the `PaulliSumOp` (rather than the legacy `WeightedPauliOperator`) and can be used as a starting point for the `to_opflow()` method.
```python
    def to_qiskit(self, mode=default_spin_mode, precomputed_XYZI_transform=None):
        """
        Creates a qiskit.aqua.operators.WeightedPauliOperator from `self` that maps the spinS
        system operator to a qubit operator.

        Args:
            mode (str):
                The mode with which the spin operators should be mapped to qubits. Should be one of
                ['lin_encoding', 'log_encoding'].

            precomputed_XYZI_transform (list):
                This list must contain exactly 4 elements corresponding to the transformations for the X, Y, Z and
                the identity operator. Each of these 4 elements must be a qiskit.aqua.operators.WeightedPauliOperator
                in `pauli` representation (i.e. a list of Pauli strings)

        Returns:
            qiskit.aqua.operators.WeightedPauliOperator
        """
        # 1. Setup
        # Get the dimension of the internal spin Hilbertspace per spin system
        dim_S = int(2 * self.S + 1)

        # Initialize an empty list, used to store the embedded qubit operators
        # used to construct the final aqua operator.
        operatorlist = []

        # 2. Generate the mapping (spin S system --> qubits)
        # Map the X, Y, Z operators of an individual spin system to the `nqubit` system.
        # The final operator will be the sum of a tensor product of these operators with
        # complex coefficients.
        if self.transformed_XYZI is None and precomputed_XYZI_transform is None:
            # if neither is given, calculate a specified transform and store it in self.transformed_XYZI
            print('Calculating encoding for operator')
            if mode == 'log_encoding':
                self.transformed_XYZI = self._logarithmic_encoding()
            elif mode == 'lin_encoding':
                self.transformed_XYZI = self._linear_encoding()
            else:
                raise UserWarning("`mode` for BaseSpinOperator to qubit transform must be one of "
                                  "['log_encoding', lin_encoding'].")

        if precomputed_XYZI_transform is None:
            spinx, spiny, spinz, identity = self.transformed_XYZI
        else:
            spinx, spiny, spinz, identity = precomputed_XYZI_transform

        # 3. Map the individual spin S systems in the register to qubits.
        # Go through the register of spin systems and construct the embedded qubit
        # operator `embed(X^nx * Y^ny * Z^nz)`. Add this embedded operator to `operatorlist`.
        for nx, ny, nz in zip(self.Sx, self.Sy, self.Sz):
            operator_on_spin_i = []
            if nx > 0:
                # construct the qubit operator embed(X^nx)
                operator_on_spin_i.append(operator_product([spinx for i in range(nx)]))
            if ny > 0:
                # construct the qubit operator embed(Y^ny)
                operator_on_spin_i.append(operator_product([spiny for i in range(ny)]))
            if nz > 0:
                # construct the qubit operator embed(Z^nz)
                operator_on_spin_i.append(operator_product([spinz for i in range(nz)]))
            if np.any([nx, ny, nz]) > 0:
                # multiply X^nx * Y^ny * Z^nz
                operator_on_spin_i = operator_product(operator_on_spin_i)
                # get rid of vanishing paulistrings
                operator_on_spin_i.chop()
                # Add the embedded qubit operator acting on spin system `i` to the operatorlist,
                # to then form a tensorproduct in the next step.
                operatorlist.append(operator_on_spin_i)
            else:
                # If nx=ny=nz=0, simply add the embedded Identity operator.
                operatorlist.append(identity)

        # `operatorlist` is now a list of (sums of pauli strings) which still need to be tensored together
        # to get the final operator

        # 4. Tensor the individual embedded operators together:
        # Extract the pauli strings from the individual site (individual spin S system) aqua operators

        paulis_per_site = [site_operator.paulis for site_operator in operatorlist]

        # Generate all possible combinations of tensor products of pauli strings which make up
        # the embedded operator. (I.e. the summand pauli strings of the final operator)
        combos = list(itertools.product(*paulis_per_site))
        # Tensor the paulis together
        tensored_paulis = [tensor_paulis(combo, multiply_coeff=self.coeff) for combo in combos]

        # Construct the final operator as a sum of these pauli strings
        final_operator = WeightedPauliOperator(paulis=tensored_paulis)

        return final_operator
```

I will also add the original code in the case of the `FermionicOperator` to #4 shortly, which can be compared against the actual implementation for that class in order to show how this code can be translated.

**Comment by mrossinek** [comment]
I am closing this because it was mostly addressed by #35. We will likely do a follow-up to that PR similarly as we did for the `FermionicOp`.
